vello/src/scene.rs

```diff
@@ -333,24 +344,340 @@ impl<'a> DrawGlyphs<'a> {
+        let run_transform = self.run.transform.to_kurbo();
+        let scale = Affine::scale_non_uniform(
+            (self.run.font_size / upem).into(),
+            (-self.run.font_size / upem).into(),
+        );
+        let colour_collection = font.color_glyphs();
+        for glyph in glyphs {
+            let colour = colour_collection
+                .get(GlyphId::new(glyph.id.try_into().unwrap()))
+                // We should fall back to rendering this as just an outline?
+                // This means creating multiple glyph runs, which is just a bit awkward
+                // If people run into this case, we can add that
+                .expect("Cannot render non-colr glyph with colr font");
+            let coords = &self.scene.encoding.resources.normalized_coords
+                [self.run.normalized_coords.clone()]
+            .to_vec();
+            let transform = run_transform
+                * Affine::translate(Vec2::new(glyph.x.into(), glyph.y.into()))
+                * scale
+                * self
+                    .run
+                    .glyph_transform
+                    .unwrap_or(Transform::IDENTITY)
+                    .to_kurbo();
+            colour
+                .paint(
+                    LocationRef::new(coords),
+                    &mut DrawColorGlyphs {
+                        scene: self.scene,
+                        cpal: &font.cpal().unwrap(),
+                        outlines: &font.outline_glyphs(),
+                        transform_stack: vec![Transform::from_kurbo(&transform)],
+                        clip_box: DEFAULT_CLIP_RECT,
+                        clip_depth: 0,
+                    },
+                )
+                .unwrap();
+        }
+    }
+}
+const BOUND: f64 = 100_000.;
+// Hack: If we don't have a clip box, we guess a rectangle we hope is big enough
+const DEFAULT_CLIP_RECT: Rect = Rect::new(-BOUND, -BOUND, BOUND, BOUND);
+
+struct DrawColorGlyphs<'a> {
+    scene: &'a mut Scene,
+    transform_stack: Vec<Transform>,
+    cpal: &'a Cpal<'a>,
+    outlines: &'a OutlineGlyphCollection<'a>,
+    clip_box: Rect,
+    clip_depth: u32,
+}
+
+impl ColorPainter for DrawColorGlyphs<'_> {
+    fn push_transform(&mut self, transform: skrifa::color::Transform) {
+        let transform = conv_skrifa_transform(transform);
+        let prior_transform = self.last_transform();
+        self.transform_stack.push(prior_transform * transform);
+    }
+
+    fn pop_transform(&mut self) {
+        self.transform_stack.pop();
+    }
+
+    fn push_clip_glyph(&mut self, glyph_id: GlyphId) {
+        let Some(outline) = self.outlines.get(glyph_id) else {
+            eprintln!("Didn't get expected outline");
+            return;
+        };
+
+        let mut path = BezPathOutline(BezPath::new());
+        let draw_settings =
+            DrawSettings::unhinted(skrifa::instance::Size::unscaled(), [].as_slice());
```

Comment:
This should use the same coords/LocationRef as passed to the paint method. Fonts can have both variable paints and variable outlines and they share the same variation space.